pkg/serverless/logs/logs_collector.go

```diff
@@ -310,6 +310,15 @@ func (lc *LambdaLogsCollector) processMessage(
 				})
 			lc.invocationEndTime = message.time
 			lc.executionContext.UpdateEndTime(message.time)
+
+			// The state is saved when a shutdown event is received. A shutdown event can occur before the
+			// runtimeDone log message is received so we save the state again to properly store the end time
```

Comment:
Just want to confirm there can't be a race condition here, where we save the context of the next invocation?

Reply:
Yup, shouldn't be the case here since we reset the `isStateSaved` bool to false at the start of the next invocation. So worse case we won't attempt to save the context